src/main.cpp

```diff
@@ -6289,10 +6289,11 @@ bool ProcessMessages(CNode* pfrom, CConnman& connman)
         uint256 hash = Hash(vRecv.begin(), vRecv.begin() + nMessageSize);
         if (memcmp(hash.begin(), hdr.pchChecksum, CMessageHeader::CHECKSUM_SIZE) != 0)
         {
-            LogPrintf("%s(%s, %u bytes): CHECKSUM ERROR expected %s was %s\n", __func__,
+            LogPrintf("%s(%s, %u bytes): CHECKSUM ERROR expected=%s was=%s peer=%d\n", __func__,
                SanitizeString(strCommand), nMessageSize,
+               HexStr(hdr.pchChecksum, hdr.pchChecksum+CMessageHeader::CHECKSUM_SIZE),
```

Comment:
hdr.pchChecksum is the checksum we are told it will be once received, therefore this is the "expected" (as the real checksum has not been calculated yet) checksum. When we receive the message, the computed checksum is the actual checksum. Is this correct?